lib/cronitor/config.rb

```diff
@@ -23,6 +31,8 @@ def configure(&block)
   self.ping_timeout = ENV.fetch('CRONITOR_PING_TIMEOUT', nil) || 5
   self.config = ENV.fetch('CRONITOR_CONFIG', nil)
   self.auto_discover_sidekiq = ENV.fetch('CRONITOR_AUTO_DISCOVER_SIDEKIQ', 'true').casecmp('true').zero? # https://github.com/cronitorio/cronitor-sidekiq
+  self.ping_url = default_ping_url
```

Comment:
This is kinda nit-picky on my part, so take it with a grain of salt, but since it's reading from the ENV var, these aren't really "defaults". Could we keep setting these the same as before, and then in the test set up where these methods are used just set the defaults correctly there?

Or, if you want to keep these methods, just have them return the string only instead of calling ENV.fetch

Reply:
I wasn't originally going to add these methods, but basically we need a method to reset the ping_url and the monitor_url from the test suite. This is because `Crontior` is like a god object, so when the first test sets the values of ping_url and the monitor_url, the second test retains the previous values. There is no way to flush or reload the `Cronitor` class.

But we need to load a fresh `Cronitor` class twice for the test suite. Once for when there are no ENV vars. And then again when there are ENV vars.

So these `default_*` methods allow me to call `self.ping_url = default_ping_url` before each test, just like they are called when the `Cronitor` class loads, so that when we change the ENV vars for the second test, the tests then behave properly and actually read the ENV vars instead of retaining the values from the first test.

We could copy the code which sets the initial ping_url and monitor_url values into the test suite and have something like below in the spec instead.

In my view this is worse, because it is not DRY and if the code in `lib/cronitor/config.rb` changes, then we are not testing the `Cronitor` class, but testing the code that is in the `before` block.

Would changing the name of the methods from `default_*` to `initial_*` be better?

```ruby
describe '#ping_url and #monitor_url' do
  let(:env_vars) { ENV }

  before do
    stub_const('ENV', env_vars)
    # We need to reset the ping_url and monitor_url since the previous spec sets them globally
    Cronitor.configure do |cronitor|
      cronitor.ping_url = ENV.fetch('CRONITOR_PING_URL', 'https://cronitor.link') # copied from lib/cronitor/config.rb
      cronitor.monitor_url = ENV.fetch('CRONITOR_MONITOR_URL', 'https://cronitor.io') # copied from lib/cronitor/config.rb
    end
  end

  context 'when custom CRONITOR_PING_URL and CRONITOR_MONITOR_URL ENV variables are set' do
    let(:env_vars) do
      ENV.to_hash.merge(
        'CRONITOR_PING_URL' => 'https://ping.com',
        'CRONITOR_MONITOR_URL' => 'https://monitor.com'
      )
    end

    it 'returns the custom Cronitor URLs' do
      expect(Cronitor.ping_url).to eq('https://ping.com')
      expect(Cronitor.monitor_url).to eq('https://monitor.com')
    end
  end

  context 'when no CRONITOR_PING_URL or CRONITOR_MONITOR_URL ENV variables are set' do
    it 'returns the default Cronitor URLs' do
      expect(Cronitor.ping_url).to eq('https://cronitor.link')
      expect(Cronitor.monitor_url).to eq('https://cronitor.io')
    end
  end
end
```

Reply:
@aflanagan